lib/rabbit_feed/event.rb

```diff
@@ -2,42 +2,52 @@ module RabbitFeed
   class Event
     include ActiveModel::Validations
 
-    attr_reader :schema, :payload
-    validates_presence_of :schema, :payload
+    attr_reader :schema, :payload, :metadata
+    validates :metadata, presence: true
+    validates :payload, length: { minimum: 0, allow_nil: false, message: 'can\'t be nil' }
 
-    def initialize schema, payload
-      @schema  = schema
-      @payload = payload
+    def initialize metadata, payload={}, schema=nil
+      @schema   = schema
+      @payload  = payload.with_indifferent_access if payload
+      @metadata = metadata.with_indifferent_access if metadata
       validate!
     end
 
+    def self.from_pre_2_0 metadata_and_payload, schema
```

Comment:
when and how client code decides to call this?. If is only internally, it should be class private.